source/extensions/filters/common/lua/wrappers.h

```diff
@@ -117,6 +117,33 @@ class MetadataMapWrapper : public BaseLuaObject<MetadataMapWrapper> {
   friend class MetadataMapIterator;
 };
 
+/**
+ * Lua wrapper for parsed fields from a X509Name
+ */
+class ParsedX509NameWrapper : public BaseLuaObject<ParsedX509NameWrapper> {
+public:
+  ParsedX509NameWrapper(const Ssl::ParsedX509Name& parsed_name) : parsed_name_{parsed_name} {}
+
+  static ExportedFunctions exportedFunctions() {
+    return {{"commonName", static_luaCommonName}, {"organizationName", static_luaOrganizationName}};
+  }
+
+private:
+  /**
+   * Returns the commonName(CN) field as a string in the X509 name. Return empty string if there
+   * is no CN field, or can't be converted to utf8 string (in case of malicious certificate).
+   */
+  DECLARE_LUA_FUNCTION(ParsedX509NameWrapper, luaCommonName);
+
+  /**
+   * Returns the organizationName(O) fields as list of strings in the X509 name. Return empty list
+   * if there is no O field, or can't be converted to utf8 string.
+   */
+  DECLARE_LUA_FUNCTION(ParsedX509NameWrapper, luaOrganizationName);
+
+  const Ssl::ParsedX509Name& parsed_name_;
```

Comment:
I wonder if it would make more sense to have this be a value instead of a reference. It's not a large struct, and then there's no concern about lifetimes, you don't need to mark-dead, etc.

Reply:
I just followed the similar code in line 314 for connection_info_. currently I only added CN and O, but eventually it should contain more fields which will increase the size and cost. I assume envoy code base want optimize code as fast as possible.